yt/visualization/plot_container.py

```diff
@@ -248,19 +249,20 @@ class PlotContainer:
     _units_config: dict
 
     def __init__(self, data_source, figure_size, fontsize):
-        from matplotlib.font_manager import FontProperties
-
         self.data_source = data_source
         self.ds = data_source.ds
         self.ts = self._initialize_dataset(self.ds)
         if is_sequence(figure_size):
             self.figure_size = float(figure_size[0]), float(figure_size[1])
         else:
             self.figure_size = float(figure_size)
-        font_path = os.path.join(
-            matplotlib.get_data_path(), "fonts", "ttf", "STIXGeneral.ttf"
-        )
-        self._font_properties = FontProperties(size=fontsize, fname=font_path)
+
+        if sys.version_info >= (3, 9):
+            font_dict = DEFAULT_FONT_PROPERTIES | {"size": fontsize}
+        else:
+            font_dict = {**DEFAULT_FONT_PROPERTIES, "size": fontsize}
```

Comment:
Why aren't you using simply
```suggestion
        font_dict = {**DEFAULT_FONT_PROPERTIES, "size": fontsize}
```
I realize the syntax with `|` is only supported on Python 3.9 and above, but I am not sure the check about the python version is worth the trouble just to use a new syntax (and it increases the programmatic complexity).

Reply:
I got enthusiastic after I managed to patch pyupgrade so that it can cleanup this kind of structure when the time comes (https://github.com/asottile/pyupgrade/pull/530), and it so happens that `|` isn't just simpler than `**..., **...`, it's also faster.
Anyway I see this as basically "free technical debt" because we have the technology to automatically get rid of it later. Now that we're on the same page about that, let me know if you changed your mind. If not I'll simplify the code as you suggested :)